src/helpers.js

```diff
@@ -192,45 +192,57 @@ export function log(context, status, ...message){
 
 /**
  * Get JSON options
- * @returns void
+ * ID priority: data-attribute > selector#id > unique id
+ * @version 0.0.1
+ * @returns {object}
  */
-export function getOptions(context){
+export function getOptions(context, defaultOptions){
+    const numeric = ['duration', 'activeSection']; // convert these props to float
+    const wrapper = context.wrapper;
+
     // options from attribute
-    let string = context.wrapper.getAttribute(context._attr.container);
+    let dataAttribute = wrapper.getAttribute(context._attr.container);
     let options = {};
 
-    if(!isJSON(string)){
-        context.id = context.options.id;
-        return;
+    // data attribute doesn't exist or not JSON format -> string
+    if(!dataAttribute || !isJSON(dataAttribute)){
+        // reassign id
+        const id = dataAttribute || wrapper.id || defaultOptions.id;
+        context.id = id;
+        defaultOptions.id = id;
+
+        return defaultOptions;
     }
 
-    // option priority: attribute > js object > default
-    options = {...JSON.parse(string)};
+    options = JSON.parse(dataAttribute);
 
-    // convert boolean string to real boolean
     for(const [key, value] of Object.entries(options)){
+        // convert boolean string to real boolean
         if(value === "false") options[key] = false;
-        if(value === "true") options[key] = true;
-        if(!isNaN(value)) options[key] = parseInt(value);
+        else if(value === "true") options[key] = true;
+        // convert string to float
+        else if(numeric.includes(key) && typeof value === 'string' && value.length > 0) options[key] = parseFloat(value);
+        else options[key] = value;
     }
 
-    // get ID
-    if(options['id'] && !isEmptyString(options['id'])){
-        context.id = options['id'];
-    }else{
-        context.id = context.options.id;
-    }
+    // reassign id
```

Comment:
Any other way to remove the duplicates?

![Screenshot 2023-01-05 at 14 50 37](https://user-images.githubusercontent.com/14942380/210729203-d46b3902-512d-470d-9c6b-705097d975dc.png)

Reply:
I have changed it a lot, more detail at https://github.com/viivue/easy-tab-accordion/pull/27/commits/0edc01397e6f56deaa3e70b34c63cdb7b9998cbf